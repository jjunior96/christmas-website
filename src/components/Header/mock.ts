import { v4 } from 'uuid';

export const HeaderMock = [
  {
    id: v4(),
    name: 'Introduction',
    href: '/'
  },
  {
    id: v4(),
    name: 'Customization',
    href: '/'
  },
  {
    id: v4(),
    name: 'Look inside',
    href: '/'
  }
];
