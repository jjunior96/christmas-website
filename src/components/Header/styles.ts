import styled, { css } from 'styled-components';

export const Container = styled.header`
  width: 100%;

  position: fixed;
  top: 0;
`;

export const Content = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.grid.container};

    display: flex;
    align-items: center;
    justify-content: space-between;

    margin: 0 auto;

    padding: ${theme.spacings.medium} 0;
  `}
`;

export const MenuContainer = styled.ul`
  display: flex;
  align-items: center;
`;

export const MenuItem = styled.li`
  ${({ theme }) => css`
    list-style: none;

    font-size: ${theme.font.sizes.xsmall};
    color: ${theme.colors.white};

    transition: color ${theme.transition.fast};

    & + li {
      margin-left: ${theme.spacings.medium};
    }

    &:hover {
      color: ${theme.colors.purple_300};
    }
  `}
`;
