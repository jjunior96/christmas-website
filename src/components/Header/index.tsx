import Link from 'next/link';

import Logo from 'components/Logo';

import { HeaderMock } from './mock';
import * as S from './styles';

const Header = () => {
  return (
    <S.Container role="menubar">
      <S.Content>
        <Logo color="white" />

        <S.MenuContainer>
          {HeaderMock.map((menu) => (
            <S.MenuItem key={menu.id}>
              <Link href={menu.href}>
                <a>{menu.name}</a>
              </Link>
            </S.MenuItem>
          ))}
        </S.MenuContainer>
      </S.Content>
    </S.Container>
  );
};

export default Header;
