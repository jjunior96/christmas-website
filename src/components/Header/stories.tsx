import { Story, Meta } from '@storybook/react/types-6-0';

import Header from '.';

export default {
  title: 'Header',
  component: Header,
  parameters: {
    backgrounds: {
      default: 'dark'
    }
  }
} as Meta;

export const Default: Story = () => <Header />;
