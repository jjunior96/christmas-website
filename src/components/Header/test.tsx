import { screen } from '@testing-library/react';

import { renderWithTheme } from '../../utils/tests/helpers';

import Header from '.';

describe('<Header />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<Header />);

    expect(screen.getByRole('menubar')).toBeInTheDocument();
    expect(screen.getByRole('menubar')).toHaveStyle({
      position: 'fixed'
    });

    expect(container.firstChild).toMatchSnapshot();
  });
});
