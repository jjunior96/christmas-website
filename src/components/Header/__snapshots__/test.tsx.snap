// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render the heading 1`] = `
.c2 {
  color: #fff;
  font-size: 4rem;
  font-weight: 600;
  font-family: 'Pacifico',cursive;
  line-height: 1;
}

.c0 {
  width: 100%;
  position: fixed;
  top: 0;
}

.c1 {
  width: 100%;
  max-width: 130rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 0 auto;
  padding: 3.2rem 0;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  list-style: none;
  font-size: 1.6rem;
  color: #fff;
  -webkit-transition: color 0.1s ease-in-out;
  transition: color 0.1s ease-in-out;
}

.c4 + li {
  margin-left: 3.2rem;
}

.c4:hover {
  color: #5855E9;
}

<header
  class="c0"
  role="menubar"
>
  <div
    class="c1"
  >
    <h2
      class="c2"
      color="white"
    >
      JR
    </h2>
    <ul
      class="c3"
    >
      <li
        class="c4"
      >
        <a
          href="/"
        >
          Introduction
        </a>
      </li>
      <li
        class="c4"
      >
        <a
          href="/"
        >
          Customization
        </a>
      </li>
      <li
        class="c4"
      >
        <a
          href="/"
        >
          Look inside
        </a>
      </li>
    </ul>
  </div>
</header>
`;
